김윤건/week1./주차 요금 계산.js

```diff
@@ -0,0 +1,53 @@
+function solution(fees, records) {
+  const answer = [];
+  const [defaultTime, defaultFee, unitTime, unitFee] = fees;
+  const timeStore = {};
+  
+  records.forEach(record => {
+      const [time, carNumber, state] = record.split(' ');
+      const [hour, min] = time.split(':');
+      const convertedTime = +hour * 60 + +min;
+      
+      if (state === 'IN') {
+          timeStore[carNumber] = {
+              ...timeStore[carNumber],
+              inTime: convertedTime,
+          };
+      } else {
+          const sumTime = timeStore[carNumber].sumTime ? timeStore[carNumber].sumTime : 0;
+          
+          timeStore[carNumber] = {
+              outTime: convertedTime,
+              sumTime: sumTime + convertedTime - timeStore[carNumber].inTime,
+              inTime: null,
+          };
```

Comment:
```suggestion
          const stayTime = convertedTime - timeStore[carNumber].inTime;
          const currentSumTime =  timeStore[carNumber].sumTime ?? 0
          
          timeStore[carNumber] = {
              outTime: convertedTime,
              sumTime:  currentSumTime + stayTime   ,
              inTime: null,
          };
```

요렇게 하면 조금 더 이해가  쉽지 않을까 생각됩니다.

Reply:
nullish에 익숙치 않아서 잘 안쓰게 됐는데 감사합니다.
제안해주신 코드가 더 가독성이 좋은 것 같습니다!